### PR TITLE
Add sleep support

### PR DESCRIPTION
**Pull request by rnestler**
Check the following:

 - [na] Breaking changes marked in commit message
 - [x] Changelog updated
 - [x] Code style cleaned (ran `make style-fix`)
 - [x] Tested on actual hardware

### CODE REVIEW

File: sps30-uart/sps30.c [hunk 1]
```diff
@@ -56,6 +59,8 @@ const char *sps_get_driver_version(void) {
 
 int16_t sps30_probe(void) {
     char serial[SPS30_MAX_SERIAL_LEN];
+    // Try to wake up, but  ignore failure if it is not in sleep mode
```
Comment (abrauchli) [inline sps30-uart/sps30.c:62]: double space
```suggestion
    // Try to wake up, but ignore failure if it is not in sleep mode
```

File: sps30-uart/sps30.c [hunk 2]
```diff
@@ -148,6 +153,25 @@ int16_t sps30_read_measurement(struct sps30_measurement *measurement) {
...
+
+    return sensirion_shdlc_xcv(SPS30_ADDR, SPS30_CMD_SLEEP, 0, NULL,
+                               0, &header, NULL);
+}
+
+int16_t sps30_wake_up(void) {
+    struct sensirion_shdlc_rx_header header;
+    int16_t ret;
+    const uint8_t data = 0xFF;
+    ret = sensirion_uart_tx(1, &data);
```
Comment (abrauchli) [inline sps30-uart/sps30.c:167]: would be style-consistent to have a blank line between declarations and code

File: sps30-uart/sps30_example_usage.c [hunk 3]
```diff
@@ -83,38 +83,56 @@ int main(void) {
...
-                   "\t%0.2f nc1.0\n"
-                   "\t%0.2f nc2.5\n"
-                   "\t%0.2f nc4.5\n"
-                   "\t%0.2f nc10.0\n"
-                   "\t%0.2f typical particle size\n\n",
-                   m.mc_1p0, m.mc_2p5, m.mc_4p0, m.mc_10p0, m.nc_0p5, m.nc_1p0,
-                   m.nc_2p5, m.nc_4p0, m.nc_10p0, m.typical_particle_size);
+            sensirion_sleep_usec(1000000); /* sleep for 1s */
+        }
+        ret = sps30_stop_measurement();
```
Comment (abrauchli) [inline sps30-uart/sps30_example_usage.c:122]: can we add a comment that the next section is about putting the sensor to sleep for one minute?

Comment (rnestler) [reply sps30-uart/sps30_example_usage.c:122]: There is a comment a few lines later that we do that.

File: sps30-uart/sps30.c (hunk 1 above)
Comment (rnestler) [reply sps30-uart/sps30.c]: Fixed.

File: sps30-uart/sps30_example_usage.c (hunk 3 above)
Comment (rnestler) [reply sps30-uart/sps30_example_usage.c:124]: I moved the comment and expanded it a bit.

File: sps30-uart/sps30.c (hunk 2 above)
Comment (rnestler) [reply sps30-uart/sps30.c:168]: Fixed.

File: sps30-uart/sps30.h [hunk 4]
```diff
@@ -115,6 +115,23 @@ int16_t sps30_stop_measurement(void);
  */
 int16_t sps30_read_measurement(struct sps30_measurement *measurement);
 
+/**
+ * sps30_sleep() - Enter sleep mode with minimum power consumption.
+ *
+ * Note that this command can only be run when the sensor is in idle
```
Comment (abrauchli) [inline sps30-uart/sps30.h:121]: add a comment that the feature is only available on FW2.0

File: sps30-uart/sps30.h [hunk 5]
```diff
@@ -115,6 +115,23 @@ int16_t sps30_stop_measurement(void);
...
+ * mode, i.e. after sps30_stop_measurement().
+ *
+ * Return:          0 on success, an error code otherwise
+ */
+int16_t sps30_sleep(void);
+
+/**
+ * sps30_wake_up() - Wake up from sleep mode and enter idle state.
+ *
```
Comment (abrauchli) [inline sps30-uart/sps30.h:130]: add a comment that the feature is only available on FW2.0

File: sps30-uart/sps30_example_usage.c [hunk 6]
```diff
@@ -83,39 +83,60 @@ int main(void) {
...
         }
-
-        sensirion_sleep_usec(1000000); /* sleep for 1s */
-    } while (1);
+        /* Stop measurement and go to sleep for 1min */
+        ret = sps30_stop_measurement();
+        if (ret) {
+            printf("Stopping measurement failed\n");
+        }
+        ret = sps30_sleep();
```
Comment (abrauchli) [inline sps30-uart/sps30_example_usage.c:128]: add a check for FW2.0?

File: sps30-uart/sps30.h (hunk 5 above)
Comment (rnestler) [reply sps30-uart/sps30.h:130]: Fixed.

File: sps30-uart/sps30.h (hunk 4 above)
Comment (rnestler) [reply sps30-uart/sps30.h]: Fixed.

File: sps30-uart/sps30_example_usage.c (hunk 6 above)
Comment (rnestler) [reply sps30-uart/sps30_example_usage.c]: Fixed. I put the sleep and wake-up functions in conditionals. Just stopping the measurement may still be a use case to save power.

**Pull request merged**